blockstore/splitstore/markset_badger.go

```diff
@@ -184,10 +195,70 @@ func (s *BadgerMarkSet) Has(c cid.Cid) (bool, error) {
 		return false, nil
 
 	default:
-		return false, xerrors.Errorf("error checking badger markset: %w", err)
+		return false, err
 	}
 }
 
+// writer holds the exclusive lock
+func (s *BadgerMarkSet) put(key string) (write bool, seqno int) {
+	s.pend[key] = struct{}{}
+	if len(s.pend) < badgerMarkSetBatchSize {
+		return false, 0
+	}
+
+	seqno = s.seqno
+	s.seqno++
+	s.writing[seqno] = s.pend
+	s.pend = make(map[string]struct{})
```

Comment:
We don't _actually_ have to do this here. Instead, we can return "full" (or even just return the number of pending items). Then, we can call `flush` with no parameters. We can then double check to see if the pending set is over the limit and, if so, flush. Otherwise, someone else wins and we walk away.

Reply:
actually, that's a pretty good idea and it will simplify things quite a bit -- no need for separate `pend` and `writing` sets this way.

Reply:
wait, we can't -- failure semantics. I'll keep it as is as the improvement would be marginal.